docs/sql-migration-guide.md

```diff
@@ -189,6 +189,69 @@ license: |
 
   - Since Spark 3.0, when using `EXTRACT` expression to extract the second field from date/timestamp values, the result will be a `DecimalType(8, 6)` value with 2 digits for second part, and 6 digits for the fractional part with microsecond precision. e.g. `extract(second from to_timestamp('2019-09-20 10:10:10.1'))` results `10.100000`.  In Spark version 2.4 and earlier, it returns an `IntegerType` value and the result for the former example is `10`.
 
+  - In Spark 3.0, datetime pattern letter `F` represents the concept of the count of days within the period of a week where the weeks are aligned to the start of the month, e.g.
+    ```sql
+    SELECT d, date_format(d, 'F') AS f
```

Comment:
Could we move the example queries into the datetime doc, then add a link here to it? IMHO most users check the doc first, maybe...

Reply:
can we use one specific date value to demonstrate the difference?

Reply:
it may not seem intuitive enough with only one date.

Reply:
I removed some cases, maybe looks better now?

Reply:
> Could we move the example queries into the datetime doc, then add a link here to it? IMHO most users check the doc first, maybe...

SGTM, I will wait until the cases look ok to you guys. Or maybe we can just make a copy there.

Reply:
Just coying them into the doc sounds okay to me.

Reply:
why one value isn't enough?
```
2020-07-30 is 4 weeks and 2 days after the first day of the month,
so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x. 
```

Reply:
> why one value isn't enough?
> 
> ```
> 2020-07-30 is 4 weeks and 2 days after the first day of the month,
> so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x. 
> ```

For 2.4, we can not say that, it's not related to the start day of month, but also need to consider the first day of week.

Reply:
```
2020-07-30 is 30 days (4 weeks and 2 days) after the first day of the month, and Thursday of the fifth week of the month. so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x.
```

We need to highlight the difference that, one is day count within the week, one is week count within the month.